src/ocaml_lsp.ml

```diff
@@ -1,65 +1,105 @@
 open! Import
 
+module Experimental_capabilities = struct
+  type t =
+    { interfaceSpecificLangId : bool
+    ; handleSwitchImplIntf : bool
+    ; handleInferIntf : bool
+    ; handleTypedHoles : bool
+    }
+
+  let default =
+    { interfaceSpecificLangId = false
+    ; handleSwitchImplIntf = false
+    ; handleInferIntf = false
+    ; handleTypedHoles = false
+    }
+
+  (** Creates [t] given a JSON of form [{ 'handleSwitchImplIntf' : true, .... }] *)
+  let t_of_json (json : Jsonoo.t) =
+    let experimental_caps_tbl = Jsonoo.Decode.(dict bool) json in
+    let has_capability cap =
+      Stdlib.Hashtbl.find_opt experimental_caps_tbl cap
```

Comment:
Why don't we use base here?

Reply:
`Jsonoo.Decode.dict` returns a `Stdlib.Hashtbl.t`, which is incompatible with `Base.Hashtbl.t`